sasl_defs.c

```diff
@@ -95,6 +96,27 @@ static int sasl_getconf(void *context, const char **path)
             }
         }
     }
+#elif defined(HAVE_SASL_CB_GETCONFPATH)
+    char buf[50];
```

Comment:
trying to avoid adding too many magic buffers like this. perhaps just make a "locations_path[]" with it pre-cut above? then we don't need the strncpy/strncat/etc stuff either.